src/VisualStudio/Core/Def/Watson/FaultReporter.cs

```diff
@@ -157,28 +157,36 @@ private static string GetDescription(Exception exception)
             try
             {
                 // walk up the stack looking for the first call from a type that isn't in the ErrorReporting namespace.
-                foreach (var frame in new StackTrace(exception).GetFrames())
+                var frames = new StackTrace(exception).GetFrames();
+
+                // On the .NET Framework, GetFrames() can return null even though it's not documented as such.
+                // At least one case here is if the exception's stack trace itself is null.
+                if (frames != null)
```

Comment:
Is this just a bug that got fixed in a later framework? Agree it's odd and looks to no longer be the case https://source.dot.net/#System.Private.CoreLib/StackTrace.cs,159

Reply:
If it wasn't a bug, it as at least deemed "behavior non grata" and changed.